### PR TITLE
fix: diagnostic folder size and humanize size

### PR DESCRIPTION
**Pull request by robertsLando**
- Fix issue of folder size calculation
- Print sizes in human friendly format
- Print folders/files that exceed a specific size. This can be set using `SIZE_LIMIT_PKG` and `FOLDER_LIMIT_PKG` env vars

### CODE REVIEW

File: prelude/diagnostic.js [hunk 1]
```diff
@@ -4,14 +4,42 @@
...
+  if (bytes === 0) {
+    return 'n/a';
+  }
+
+  const i = Math.floor(Math.log(bytes) / Math.log(1024));
+
+  if (i === 0) {
+    return `${bytes} ${sizes[i]}`;
+  }
+
```
Comment (erossignon) [inline prelude/diagnostic.js:19]: Strictly speaking we should add a boundary checking here.
If someone passed a number of bytes that's larger than what 'TB' can represent, the function would try to access an index in the sizes array that doesn't exist, which would result in undefined.
But this is unlikely that someone will try to pack a file that is greater than a TB.

File: prelude/diagnostic.js [hunk 2]
```diff
@@ -4,14 +4,42 @@
...
+  return `${(bytes / 1024 ** i).toFixed(1)} ${sizes[i]}`;
+}
+
 (function installDiagnostic() {
   const fs = require('fs');
   const path = require('path');
   const win32 = process.platform === 'win32';
 
+  // set a size limit of 5MB
```
Comment (erossignon) [inline prelude/diagnostic.js:28]: The comment is lying.

**Review by erossignon — APPROVED** [review]

File: prelude/diagnostic.js (hunk 1 above)
Comment (robertsLando) [reply prelude/diagnostic.js:19]: > But this is unlikely that someone will try to pack a file that is greater than a TB.

Yeah I wouldn't expext that BTW js would just return `undefined` there, so something that would throw anyway

File: prelude/diagnostic.js (hunk 2 above)
Comment (robertsLando) [reply prelude/diagnostic.js]: Agree, removed both comments

**Comment by robertsLando** [comment]
@erossignon FYI giving that I don't receive any feedback from Vercel anymore, I created my own forks of both pkg and pkg-fetch:

https://github.com/yao-pkg/pkg-fetch
https://github.com/yao-pkg/pkg

**Pull request closed without merging**